network/src/peer_store/addr_manager.rs

```diff
@@ -49,17 +50,21 @@ impl AddrManager {
             let j = rng.gen_range(i, self.random_ids.len());
             self.swap_random_id(j, i);
             let addr_info: AddrInfo = self.id_to_info[&self.random_ids[i]].to_owned();
-            let is_unique_ip = duplicate_ips.insert(addr_info.ip_port.ip);
-            // A trick to make our tests work
-            // TODO remove this after fix the network tests.
-            let is_test_ip =
-                addr_info.ip_port.ip.is_unspecified() || addr_info.ip_port.ip.is_loopback();
-            if (is_test_ip || is_unique_ip) && !addr_info.is_terrible(now_ms) && filter(&addr_info)
-            {
-                addr_infos.push(addr_info);
-            }
-            if addr_infos.len() == count {
-                break;
+            if let Some(socket_addr) = multiaddr_to_socketaddr(&addr_info.addr) {
```

Comment:
Why do we need to assume there are invalid addresses in the manager? How to deal with these addresses? I think to check the format of the address in the entry of the peer store is better than just ignore them.

Reply:
it's not assumption, we have checked the format of address in the entry of the peer store already, it's just a different style of code: we can use expect here, but I prefer `if let Some()`

Reply:
`if let Some()` style often makes errors implicit. However, it depends on you, I do not insist on this.